src/platform_linux/platform.h

```diff
@@ -356,24 +357,32 @@ extern platform_heap_id Heap_id;
  * the difference that the alignment is caller-specified.
  *
  * Parameters:
+ *	mf  - platform_memfrag *, to return memory allocation information.
  *	hid - Platform heap-ID to allocate memory from.
  *	a   - Alignment needed for allocated memory.
  *	v   - Structure to allocate memory for.
  *	n   - Number of bytes of memory to allocate.
  */
-#define TYPED_ALIGNED_MALLOC(hid, a, v, n)                                     \
+#define TYPED_ALIGNED_MALLOC_MF(mf, hid, a, v, n)                              \
    ({                                                                          \
       debug_assert((n) >= sizeof(*(v)));                                       \
       (typeof(v))platform_aligned_malloc(                                      \
-         hid, (a), (n), STRINGIFY(v), __func__, __FILE__, __LINE__);           \
+         (mf), hid, (a), (n), STRINGIFY(v), __func__, __FILE__, __LINE__);     \
    })
-#define TYPED_ALIGNED_ZALLOC(hid, a, v, n)                                     \
+
+#define TYPED_ALIGNED_MALLOC(hid, a, v, n)                                     \
+   TYPED_ALIGNED_MALLOC_MF(&memfrag_##v, hid, a, v, n)
```

Comment:
I hate this hard-coding of this naming convention, and I think it doesn't solve an important problem.

Let me propose something that might be more useful.

```
#define TYPED_ALIGNED_MALLOC_AUTO(hid, a, v, n)                                     \
  platform_memfrag v_##memfrag __attribute__(cleanup(platform_free));
  TYPED_ALIGNED_MALLOC_MF(&v_##memfrag, hid, a, v, n)
```

This allocates memory that will automatically get freed when the function exits the scope of the allocation.

Furthermore, it alleviates the user of the responsibility to declare a memfrag at all.